### ISSUE TITLE
Calling a nil func will sometimes fail with a SIGILL

### ISSUE THREAD

**Issue by jadr2ddude** [tinygo-org/tinygo#623]
Minimal demo to reproduce:

```
package main

import "sync"

var a = &sync.Pool{
        New: func() interface{} {
                return 0
        },
}
var b = &sync.Pool{
        New: func() interface{} {
                return "hello"
        },
}

var c sync.Pool

func main() {
        println(a.Get())
        println(b.Get())
        println(c.Get())
}
```

In this demo, the first two attempts print things, and then the third crashes with an illegal instruction. The third is supposed to panic, but the nil panic somehow gets misoptimized into an `llvm.trap()` call.

I will try to see if I can get a more specific demo.

**Comment by jadr2ddude** [comment]
And this demo actually shouldnt be right :smile:

**Comment by jadr2ddude** [comment]
Fixed in #625
